client/dfclient/include/lerp_component.hpp

```diff
@@ -0,0 +1,41 @@
+#include <ncine/Sprite.h>
+
+#include "../../../common/constants.hpp"
+#include "../../../common/deadfish_generated.h"
+
+class LerpComponent {
+    ncine::Sprite* sprite;
+	ncine::Vector2f prevPosition = {};
+	ncine::Vector2f currPosition = {};
+	float prevRotation = 0.f;
+	float currRotation = 0.f;
+
+public:
+    inline LerpComponent() {}
+    inline void bind(ncine::Sprite* aSprite) { sprite = aSprite; }
+	inline void setupLerp(const float x, const float y, const float angle, bool firstUpdate) {
```

Comment:
Would this work if we converted `firstUpdate` into a field of the `LerpComponent`?

Reply:
Yes it would, actually even better. Thanks for the tip, will refactor